javalibpst/src/main/java/fr/gouv/vitam/tools/javalibpst/LZFu.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2010 Richard Johnson & Orin Eman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * ---
+ *
+ * This file is part of javalibpst.
+ *
+ * javalibpst is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * javalibpst is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with javalibpst. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package fr.gouv.vitam.tools.javalibpst;
+
+import java.io.UnsupportedEncodingException;
+
+/**
+ * An implementation of the LZFu algorithm to decompress RTF content
+ * 
+ * @author Richard Johnson
+ */
+public class LZFu {
+
+    public static final String LZFU_HEADER = "{\\rtf1\\ansi\\mac\\deff0\\deftab720{\\fonttbl;}{\\f0\\fnil \\froman \\fswiss \\fmodern \\fscript \\fdecor MS Sans SerifSymbolArialTimes New RomanCourier{\\colortbl\\red0\\green0\\blue0\n\r\\par \\pard\\plain\\f0\\fs20\\b\\i\\u\\tab\\tx";
+
+    public static String decode(final byte[] data) throws PSTException {
+
+        @SuppressWarnings("unused")
+        final int compressedSize = (int) PSTObject.convertLittleEndianBytesToLong(data, 0, 4);
+        final int uncompressedSize = (int) PSTObject.convertLittleEndianBytesToLong(data, 4, 8);
+        final int compressionSig = (int) PSTObject.convertLittleEndianBytesToLong(data, 8, 12);
+        @SuppressWarnings("unused")
+        final int compressedCRC = (int) PSTObject.convertLittleEndianBytesToLong(data, 12, 16);
+
+        if (compressionSig == 0x75465a4c) {
+            // we are compressed...
+            final byte[] output = new byte[uncompressedSize];
+            int outputPosition = 0;
+            final byte[] lzBuffer = new byte[4096];
+            // preload our buffer.
+            try {
+                final byte[] bytes = LZFU_HEADER.getBytes("US-ASCII");
```

Comment:
use StandardCharsets.US_ASCII

Reply:
The Javalibpst library was previously in the repository [https://github.com/ProgrammeVitam/java-libpst-origin](https://github.com/ProgrammeVitam/java-libpst-origin) with hopes of contributing to its evolution. Since it hasn't been updated for years, I preferred to integrate it into the project to make the necessary modifications for multi-threading, which required intensive testing (the code is poorly documented and handles complex Microsoft format concepts). The issues noted existed in the original code which is functional. I don't consider it necessary to fix these issues in this merge. I'm willing to consider it in another merge.